classes/Forms.js

```diff
@@ -34,17 +34,26 @@ module.exports = class Forms extends OneBlinkAPI {
     externalId /* : ?mixed */,
     preFillData /* : ?mixed */
   ) /* : Promise<{ expiry: string, formUrl: string }> */ {
-    const organisationsResults = await super.searchRequest(`/organisations`)
     if (typeof formId !== 'number') {
       throw new TypeError('Must supply "formId" as a number')
     }
-    if (externalId && typeof externalId !== 'string') {
-      throw new TypeError('Must supply "externalId" as a string or not at all')
+
+    const formResults = await super.getRequest(`/forms/${formId}`)
+
+    if (!formResults || !formResults.formsAppIds.length) {
+      throw new Error('Access Denied. The "accessKey" and "secretKey" provided may have been revoked.')
     }
-    if (!organisationsResults.organisations || !organisationsResults.organisations.length) {
+
+    const appResults = await super.getRequest(`/forms-apps/${formResults.formsAppIds[0]}`)
+
+    if (!appResults || !appResults.hostname) {
       throw new Error('Access Denied. The "accessKey" and "secretKey" provided may have been revoked.')
     }
 
+    if (externalId && typeof externalId !== 'string') {
+      throw new TypeError('Must supply "externalId" as a string or not at all')
+    }
```

Comment:
Can we move this back up to the top to avoid unnecessary network requests if the arguments passed are invalid.

Reply:
yep yep